Google.Api.Gax.Grpc/ChannelBaseExtensions.cs

```diff
@@ -0,0 +1,41 @@
+﻿/*
+ * Copyright 2022 Google Inc. All Rights Reserved.
+ * Use of this source code is governed by a BSD-style
+ * license that can be found in the LICENSE file or at
+ * https://developers.google.com/open-source/licenses/bsd
+ */
+
+using Grpc.Core;
+using Grpc.Net.Client;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Threading.Tasks;
+
+namespace Google.Api.Gax.Grpc;
+
+/// <summary>
+/// Extension methods for <see cref="ChannelBase"/>.
+/// </summary>
+public static class ChannelBaseExtensions
+{
+    /// <summary>
+    /// Shuts down a channel semi-synchronously. This method initially calls <see cref="IDisposable.Dispose"/>
+    /// if the channel implements <see cref="IDisposable"/> (e.g. in the case of <see cref="GrpcChannel"/>)
+    /// and then calls <see cref="ChannelBase.ShutdownAsync"/>. This method does not wait for the task
+    /// to complete, but observes any exceptions (whether the tasks is faulted or canceled), optionally logging
+    /// them to <paramref name="logger"/>.
+    /// </summary>
+    /// <param name="channel">The channel to shut down.</param>
+    /// <param name="logger">An optional logger to record any errors during asynchronous shutdown.</param>
+    public static void Shutdown(this ChannelBase channel, ILogger logger = null)
+    {
+        GaxPreconditions.CheckNotNull(channel, nameof(channel));
+        if (channel is IDisposable disposable)
+        {
+            disposable.Dispose();
+        }
+        channel.ShutdownAsync().ContinueWith(
+            task => logger?.LogWarning(task.Exception, task.IsCanceled ? "Channel shutdown canceled" : "Channel shutdown failed"),
```

Comment:
I think that if the logger is null, then the task Exception may be marked as unobserved and possibly provoke an UnobservedTaskException down the line. Maybe just access the task.Exception regardless of whether there's a logger or not?

Reply:
Yes, will do.